src/overlays/actors/ovl_Dm_Al/z_dm_al.c

```diff
@@ -30,24 +27,133 @@ const ActorInit Dm_Al_InitVars = {
     (ActorFunc)DmAl_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_0600A0D8;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/func_80C1BD90.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/func_80C1BDD8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/DmAl_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/DmAl_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/DmAl_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/func_80C1C028.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/func_80C1C064.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/func_80C1C11C.s")
+static AnimationInfoS sAnimationInfos[] = {
+    { &object_al_Anim_00DBE0, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Al/DmAl_Draw.s")
+s32 func_80C1BD90(DmAl* this, s32* animationIndex) {
+    s32 ret = 0;
+
+    if (animationIndex != this->animationIndex) {
+        this->animationIndex = animationIndex;
+        ret = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfos, animationIndex);
+    }
+    return ret;
+}
+
+void func_80C1BDD8(DmAl* this, PlayState* play) {
+    s32 D_80C1C280[] = { 0, 0, 0, 0, 0 };
+    u16 action;
+    s32 actionIndex;
+
+    if (play->csCtx.state != 0) {
+        if (this->unk45C == 0) {
+            this->action = 0xFF;
+            this->unk45C = 1;
+            this->animationIndex2 = this->animationIndex;
+        }
+        if (Cutscene_CheckActorAction(play, 0x232)) {
+            actionIndex = Cutscene_GetActorActionIndex(play, 0x232);
+            action = play->csCtx.actorActions[actionIndex]->action;
+            if (this->action != (u8)action) {
+                this->action = action;
+                func_80C1BD90(this, D_80C1C280[action]);
+            }
+            Cutscene_ActorTranslateAndYaw(&this->actor, play, actionIndex);
+        }
+    } else if (this->unk45C != 0) {
+        this->unk45C = 0;
+        func_80C1BD90(this, this->animationIndex2);
+    }
+}
+
+void DmAl_Init(Actor* thisx, PlayState* play) {
+    DmAl* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gMadameAromaSkel, NULL, this->jointTable, this->morphTable,
+                       OBJECT_AL_LIMB_MAX);
+    this->animationIndex = -1;
+    func_80C1BD90(this, NULL);
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = func_80C1BDD8;
+}
+
+void DmAl_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void DmAl_Update(Actor* thisx, PlayState* play) {
+    DmAl* this = THIS;
+
+    this->actionFunc(this, play);
+    SkelAnime_Update(&this->skelAnime);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 30.0f, 12.0f, 0.0f, 4);
+}
+
+s32 func_80C1C028(PlayState* play, s32 limbIndex, Gfx** gfx, Vec3f* rot, Vec3s* pos, Actor* thisx) {
```

Comment:
```suggestion
s32 func_80C1C028(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* rot, Vec3s* pos, Actor* thisx) {
```
These should have the same naming conventions as the typedefs in `z64animation.h`